action/action-util/src/main/java/com/powsybl/action/util/GeneratorScalable.java

```diff
@@ -48,47 +63,89 @@ public void reset(Network n) {
     }
 
     @Override
-    public double maximumValue(Network n) {
+    public double maximumValue(Network n, ScalingConvention scalingConvention) {
         Objects.requireNonNull(n);
+        Objects.requireNonNull(scalingConvention);
 
         Generator g = n.getGenerator(id);
-        return g != null ? g.getMaxP() : 0;
+        if (g != null) {
+            return scalingConvention == GENERATOR ? Math.min(g.getMaxP(), maxValue) : -Math.max(g.getMinP(), minValue);
+        } else {
+            return 0;
+        }
+
     }
 
     @Override
-    public void listGenerators(Network n, List<Generator> generators, List<String> notFoundGenerators) {
+    public double minimumValue(Network n, ScalingConvention scalingConvention) {
         Objects.requireNonNull(n);
-        Objects.requireNonNull(generators);
 
         Generator g = n.getGenerator(id);
         if (g != null) {
-            generators.add(g);
+            return scalingConvention == GENERATOR ? Math.max(g.getMinP(), minValue) : -Math.min(g.getMaxP(), maxValue);
         } else {
-            if (notFoundGenerators != null) {
-                notFoundGenerators.add(id);
-            }
+            return 0;
```

Comment:
-Double.MAX_VALUE ?

Reply:
No, if the generator exist, you can shift 0 on the Scalable, so minValue = maxValue = initialValue = 0

Reply:
OK